### PR TITLE
certificate routing and template context

### PR DESCRIPTION
**Pull request by HamzaIbnFarooq**
#### What are the relevant tickets?
Fixes https://github.com/mitodl/bootcamp-ecommerce/issues/1107

#### What's this PR do?
Adds routing for certificates and places important context for certificate templates

#### How should this be manually tested?
Create a `certificate index page` under `Home` with `slug=certificate`
Create a `certificate template` through CMS, under any `bootcamp-run`
Go to `admin panel` and create a `bootcamp-run-certificate` of that `bootcamp-run`
Go to `/certificate/uuid-of-certificate` to view certificate

#### Screenshots (if appropriate)
Certificate Fields in CMS:
image

Rendered Template (Theming is not done yet, just showing the important data required in template):
image

### CODE REVIEW

File: cms/models.py [hunk 1]
```diff
@@ -845,6 +858,60 @@ def serve(self, request, *args, **kwargs):
...
+        You can only create one of these pages under the home page.
+        The parent is limited via the `parent_page_type` list.
+        """
+        return (
+            super().can_create_at(parent)
+            and not parent.get_children().type(cls).exists()
+        )
+
+    @route(r"^([A-Fa-f0-9-]+)/?$")
+    def course_certificate(
```
Comment (asadiqbal08) [inline cms/models.py:884]: I guess, it would be `bootcamp_certificate`

File: cms/models.py [hunk 2]
```diff
@@ -845,6 +858,60 @@ def serve(self, request, *args, **kwargs):
...
+        self, request, uuid, *args, **kwargs
+    ):  # pylint: disable=unused-argument
+        """
+        Serve a course certificate by uuid
```
Comment (asadiqbal08) [inline cms/models.py:888]: nit: bootcamp certificate

**Review by asadiqbal08 — APPROVED** [review]
just minor things.

File: cms/models.py (hunk 1 above)
Comment (HamzaIbnFarooq) [reply cms/models.py]: Updated.

File: cms/models.py (hunk 2 above)
Comment (HamzaIbnFarooq) [reply cms/models.py]: Updated.

**Pull request merged**